src/utils/localStorage.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+function getStorageValue(key: string, defaultValue: unknown) {
+  // getting stored value
+  const saved = localStorage.getItem(key);
+  if (typeof saved === "string") {
```

Comment:
the use of 'saved', 'initial' and 'key' here could have more specific names to help the reader better understand what localStorage is specifically being used for. We can see its for items, but what items (resource, tag, type etc)?